### PR TITLE
py3++: test/mitmproxy/test_server

### PR DESCRIPTION
**Pull request by dufferzafar**

### CODE REVIEW

File: test/mitmproxy/tservers.py [hunk 1]
```diff
@@ -148,7 +150,7 @@ def pathod(self, spec, sni=None):
             Constructs a pathod GET request, with the appropriate base and proxy.
         """
         p = self.pathoc(sni=sni)
-        spec = spec.encode("string_escape")
```
Comment (dufferzafar) [inline test/mitmproxy/tservers.py]: ^

File: test/mitmproxy/tservers.py [hunk 2]
```diff
@@ -43,7 +45,7 @@ def clear_log(self):
         self.tlog = []
 
     def add_event(self, message, level=None):
-        self.tlog.append(message)
+        self.tlog.append(strutils.always_bytes(message))
```
Comment (dufferzafar) [inline test/mitmproxy/tservers.py]: ^

**Comment by dufferzafar** [comment]
Not all tests have yet passed. So this is WIP!

File: test/mitmproxy/tservers.py (hunk 2 above)
Comment (mhils) [inline test/mitmproxy/tservers.py]: I think log events should always be text, not bytes.

File: test/mitmproxy/tservers.py (hunk 1 above)
Comment (mhils) [inline test/mitmproxy/tservers.py]: If it does not cause any issues, let's remove the line entirely.

File: netlib/http/request.py [hunk 3]
```diff
@@ -28,7 +28,7 @@ def __init__(self, first_line_format, method, scheme, host, port, path, http_ver
 
         self.first_line_format = first_line_format
         self.method = method
-        self.scheme = scheme
+        self.scheme = scheme or b''
```
Comment (dufferzafar) [inline netlib/http/request.py]: Made this change to fix these errors: https://travis-ci.org/mitmproxy/mitmproxy/jobs/140021906#L342-L365

Comment (Kriechi) [inline netlib/http/request.py]: How does this relate? Why/where is `scheme` not set?

Comment (dufferzafar) [inline netlib/http/request.py]: I think I linked the wrong lines. [These](https://travis-ci.org/mitmproxy/mitmproxy/jobs/140021906#L367-L384) are the ones!

File: test/mitmproxy/tservers.py [hunk 4]
```diff
@@ -43,7 +45,7 @@ def clear_log(self):
         self.tlog = []
 
     def add_event(self, message, level=None):
-        self.tlog.append(message)
+        self.tlog.append(strutils.native(message, "utf8"))
```
Comment (mhils) [inline test/mitmproxy/tservers.py]: Looks like this fixes symptoms, not causes. Where do we get binary events from?

**Comment by mhils** [comment]
Fixed & Merged! :smiley:

**Pull request closed without merging**